DatadogInternal/Tests/Telemetry/TelemetryTests.swift

```diff
@@ -5,267 +5,179 @@
+        XCTAssertEqual(telemetry.messages.count, 1)
+    }
 
+    func testSendingErrorTelemetry_whenNoKindAndNoStack() throws {
         // When
-        #sourceLocation(file: "File.swift", line: 2)
-        telemetry.error(nsError)
+        #sourceLocation(file: "File.swift", line: 1)
+        telemetry.error("error message", kind: nil, stack: nil)
         #sourceLocation()
 
         // Then
-        XCTAssertEqual(telemetry.error?.id, "File.swift:2:error description")
-        XCTAssertEqual(telemetry.error?.message, "error description")
-        XCTAssertEqual(telemetry.error?.kind, "custom-domain - 10")
-        XCTAssertEqual(
-            telemetry.error?.stack,
-            """
-            Error Domain=custom-domain Code=10 "error description" UserInfo={NSLocalizedDescription=error description}
-            """
-        )
-
-        // When
-        telemetry.error("swift error", error: swiftError)
-        // Then
-        XCTAssertEqual(telemetry.error?.message, #"swift error - SwiftError(description: "error description")"#)
-
-        // When
-        telemetry.error("ns error", error: nsError)
-        // Then
-        XCTAssertEqual(telemetry.error?.message, "ns error - error description")
+        let error = try XCTUnwrap(telemetry.messages.firstError())
+        XCTAssertEqual(error.id, "\(moduleName())/File.swift:1:error message")
+        XCTAssertEqual(error.message, "error message")
+        XCTAssertEqual(error.kind, "\(moduleName())/FileError")
+        XCTAssertEqual(error.stack, "\(moduleName())/File.swift:1")
+        XCTAssertEqual(telemetry.messages.count, 1)
     }
 
-    func testTelemetryConfiguration() {
+    func testSendingErrorTelemetry_withSwiftError() throws {
         // Given
-        let expectedConfiguration: ConfigurationTelemetry = .mockRandom()
-
-        let telemetry = TelemetryTest()
+        struct SwiftError: Error {
+            let description = "error description"
+        }
+        let swiftError = SwiftError()
 
         // When
-        telemetry.applyConfiguration(configuration: expectedConfiguration)
+        telemetry.error(swiftError)
+        telemetry.error("custom message", error: swiftError)
 
         // Then
-        XCTAssertEqual(telemetry.configuration, expectedConfiguration)
+        let errors = telemetry.messages.compactMap({ $0.asError })
+        XCTAssertEqual(telemetry.messages.count, 2)
+        XCTAssertEqual(errors[0].message, #"SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[0].kind, "SwiftError")
+        XCTAssertEqual(errors[0].stack, #"SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[1].message, #"custom message - SwiftError(description: "error description")"#)
+        XCTAssertEqual(errors[1].kind, "SwiftError")
+        XCTAssertEqual(errors[1].stack, #"SwiftError(description: "error description")"#)
     }
 
-    func testTelemetryConfigurationMerge() {
```

Comment:
[This test](https://github.com/DataDog/dd-sdk-ios/blob/876787a4b056350b8ace92501d53edcd910bd664/DatadogInternal/Tests/Telemetry/TelemetryTests.swift#L125-L139) was removed as it was not testing SDK logic. It was calling the mock and asserting if the mock is updated.